### ISSUE TITLE
Docs in GrpcChannelProperties invalid about `dns` scheme and have minor other problems

### ISSUE THREAD

**Issue by FyiurAmron** [grpc-ecosystem/grpc-spring#1024]
**The context**

https://github.com/grpc-ecosystem/grpc-spring/blob/master/grpc-client-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/client/config/GrpcChannelProperties.java#L101 (and the following lines, up to L104 inclusive) describe the `dns` scheme for channel URLs, seemingly to reflect official gRPC docs from https://github.com/grpc/grpc/blob/master/doc/naming.md

**The bug**

Official docs define the scheme as `dns:[//authority/]host[:port]` , so the URLs should be either e.g.
 1. `dns:foobar:9090`
 2. `dns:///foobar:9090` (using default `localhost` authority, equivalent to 1.)
 3. `dns://10.0.0.1/foobar:9090` (using server from `10.0.0.1` for resolution)

However current docs describe it as follows:

     * <li>{@code dns:/localhost (might refer to the IPv4 or the IPv6 address or both, dependent on the system
     * configuration, it does not check whether there is actually someone listening on that network interface)}</li>
     * <li>{@code dns:/example.com}</li>
     * <li>{@code dns:/example.com:9090}</li>
     * <li>{@code dns:///example.com:9090}</li>

, which contradicts the official gRPC docs in the three first cases, has the `@code` too large in the first case (shouldn't wrap the parenthesized remark, it's not code), and also doesn't explain the `///` idiom at all (official docs do it for a good reason, it's really easy to misunderstand it).

**Additional context**

This doc was added in https://github.com/grpc-ecosystem/grpc-spring/commit/900c651c693aeb235f8722261abc1484f274bfc0 . If "pull requests are welcome", I'll gladly fix this one and do an PR for it.

**Comment by ST-DDT** [comment]
PRs are welcome.

- `scheme://location` did not work as expected when I tested it back then (or wasn't implemented?).
- `scheme:location` didn't work either.
- schema vs scheme might be a typo
- too large code -> true
- `scheme:[//authority]/path` -> true
- host vs path -> it acts as a host/name so IMO that is more appropriate here
- `{@code dns:/}` -> maybe `{@code dns:/some-address}`
-  this makes the official docs invalid -> please address that upstream

**Comment by FyiurAmron** [comment]
> * `scheme://location` did not work as expected when I tested it back then (or wasn't implemented?).

`static://location` does most certainly work. Two slashes won't work for `dns` though, and the template was given as to relevant for all of the cases, I guess.

> this makes the official docs invalid -> please address that upstream
yeah, I'll throw a bug report at the official repo on Monday.

**Comment by ST-DDT** [comment]
> > * `scheme://location` did not work as expected when I tested it back then (or wasn't implemented?).
> 
> `static://location` does most certainly work. Two slashes won't work for example for `dns` though (and the template was given as to relevant for all of the cases, I guess)
> 
> > * `scheme:location` didn't work either.
> 
> AFAIR it works for `unix` relative path scheme (+ the remark above).

I meant specifically the `dns` scheme.

**Comment by FyiurAmron** [comment]
> > > * `scheme://location` did not work as expected when I tested it back then (or wasn't implemented?).
> > 
> > 
> > `static://location` does most certainly work. Two slashes won't work for example for `dns` though (and the template was given as to relevant for all of the cases, I guess)
> > > * `scheme:location` didn't work either.
> > 
> > 
> > AFAIR it works for `unix` relative path scheme (+ the remark above).
> 
> I meant specifically the `dns` scheme.

yeah, my point here is that currently the docs say `"The target uri must be in the format: * {@code schema:[//[authority]][/path]}."` etc. not about just the `dns` scheme, but in general, about all of them, that's why this format simply isn't right IMVHO. It's only a format valid for _some_ of the schemes, yet the docs say "must" when speaking about all of them.

BTW,
>host vs path -> it acts as a host/name so IMO that is more appropriate here

so should we call that "host" or "path" finally? ATM it says `/path`

**Comment by ST-DDT** [comment]
> BTW,
> 
> > host vs path -> it acts as a host/name so IMO that is more appropriate here
> 
> so should we call that "host" or "path" finally? ATM it says `/path`

Sorry, for being confusing.

I meant we should use `path` when speaking about unspecific URIs as per spec.
However when we refer specifically to the `dns` scheme, then we should call it `host`.

**Comment by FyiurAmron** [comment]
Bug in upstream doc reported at https://github.com/grpc/grpc/issues/35539
